js/bitopro.js

```diff
@@ -0,0 +1,814 @@
+            'quoteVolume': undefined,
+            'info': ticker,
+        }, market, false);
+    }
+
+    async fetchTicker (symbol, params = {}) {
+        const market = this.market (symbol);
+        const request = {
+            'pair': market['id'],
+        };
+        const response = await this.publicGetTickersPair (this.extend (request, params));
+        const ticker = this.safeValue (response, 'data', {});
+        //
+        //     {
+        //         "data":{
+        //             "pair":"btc_twd",
+        //             "lastPrice":"1182449.00000000",
+        //             "isBuyer":false,
+        //             "priceChange24hr":"-1.99",
+        //             "volume24hr":"9.13089740",
+        //             "high24hr":"1226097.00000000",
+        //             "low24hr":"1181000.00000000"
+        //         }
+        //     }
+        //
+        return this.parseTicker (ticker, market);
+    }
+
+    async fetchTickers (symbols = undefined, params = {}) {
+        const response = await this.publicGetTickers ();
+        const tickers = this.safeValue (response, 'data', []);
+        //
+        //     {
+        //         "data":[
+        //             {
+        //                 "pair":"xrp_twd",
+        //                 "lastPrice":"21.26110000",
+        //                 "isBuyer":false,
+        //                 "priceChange24hr":"-6.53",
+        //                 "volume24hr":"102846.47084802",
+        //                 "high24hr":"23.24460000",
+        //                 "low24hr":"21.13730000"
+        //             }
+        //         ]
+        //     }
+        //
+        return this.parseTickers (tickers, symbols);
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        const timestamp = this.milliseconds ();
+        await this.loadMarkets ();
+        const request = {
+            'pair': this.marketId (symbol),
+        };
+        if (limit !== undefined) {
+            request['limit'] = limit;
+        }
+        const response = await this.publicGetOrderBookPair (this.extend (request, params));
+        //
+        //     {
+        //         "bids":[
+        //             {
+        //                 "price":"1175271",
+        //                 "amount":"0.00022804",
+        //                 "count":1,
+        //                 "total":"0.00022804"
+        //             }
+        //         ],
+        //         "asks":[
+        //             {
+        //                 "price":"1176906",
+        //                 "amount":"0.0496",
+        //                 "count":1,
+        //                 "total":"0.0496"
+        //             }
+        //         ]
+        //     }
+        //
+        return this.parseOrderBook (response, symbol, timestamp, 'bids', 'asks', 'price', 'amount');
+    }
+
+    parseTrade (trade, market) {
+        //
+        //     {
```

Comment:
we can remove the data wrapper (" data: []..)" from the sample, parsers already receive a single object 🙂

Reply:
Yep, you're right.